Services/IotHub/IotHubConnectionStringManager.cs

```diff
@@ -216,11 +220,11 @@ private void ValidateConnectionString(string connectionString)
 
         private async Task TestIoTHubReadPermissionsAsync(string connectionString)
```

Comment:
the code is not testing the conn string passed

Reply:
the connection string itself is tested in a separate call, this.ValidateConnectionString(connectionString).

Reply:
this method receives in input a connection string to test (`connectionString`), where is is the variable used?

Reply:
For instance the two methods `TestIoTHubWritePermissionsAsync` and `TestIoTHubReadPermissionsAsync` which seems similar, work in 2 very different way. Am I missing something?

Reply:
the connection string passed to TestIoTHubReadPermissionsAsync has the following code in the try block:
                this.registryManager.Init(connectionString);
This is how the connectionString is used.

Reply:
Previously both TestIoTHubReadPermissionsAsync and TestIoTHubWritePermissionsAsync were testing read and write by trying to read devices and add a new device to devices respectively. But the getDevices function used to read devices was deprecated. So, we have to replace that with a different one, which is what we are trying to do through this PR

Reply:
I see the code has been added in the last commit, thanks